bonus_guides/I_sending_email.md

```diff
@@ -0,0 +1,333 @@
+Sending email from a Phoenix application is really easy. Before we begin, we'll need an account with mailgun.com - we won't actually be able to send mail without it. Once we have an account, though, the rest will be straightforward.
+
+In order to sign up for a mailgun.com account, just head over to `http://www.mailgun.com/`, click on the "sign up" link, and fill out the form. They have a generous number of free emails per month, so we can get going with a free account.
+
+Once we have an account, we'll get a sandbox through which we can send mail. The url of that sandbox will be our domain unless we choose to create a custom one through mailgun.com.
+
+Now that we have an account, we'll need to add `mailgun` as a dependency to our project. We'll do that in the `deps/0` function in `mix.exs`.
+
+```elixir
+defp deps do
+  [{:phoenix, "~> 0.17"},
+   {:phoenix_ecto, "~> 0.9"},
+   {:postgrex, ">= 0.0.0"},
+   {:phoenix_html, "~> 2.0"},
+   {:phoenix_live_reload, "~> 0.6", only: :dev},
+   {:cowboy, "~> 1.0"},
+   {:mailgun, "~> 0.1.1"}]
+end
+```
+
+Next, we'll need to run `mix deps.get` to bring the mailgun package into our application.
+
+We'll also need to configure our mailgun domain and api key in `config/config.ex`.
+
+The mailgun domain will be a full url, something like this `https://api.mailgun.net/v3/sandbox-our-domain.mailgun.org`. The key will be a long string - "key-another-long-string". For security reasons, it's important to not commit these values to a public source code repository, so we'll use environment variables set in our system to bring them into our configuration.
+
+Here's how that would look in `config/config.exs`
+
+```elixir
+config :hello_phoenix,
+       mailgun_domain: System.get_env("MAILGUN_DOMAIN")
+       mailgun_key: System.get_env("MAILGUN_API_KEY")
```

Comment:
I thought about this last night and my sense is that we need to recommend something like a parallel "secrets" system for dev - `dev.secret.exs`, import it into `dev.exs`, and put `dev.secret.exs` in `.gitignore`.

Of course, this is in addition to putting the prod mailgun credentials in `prod.secret.exs` as well so that it will work in production.

Without `dev.secret.exs`, I don't see how we can keep the credentials out of source control and also out of environment variables for the dev environment.

Alternately, in order to remove duplication, we could suggest a `config.secret.exs`, import that into `config.exs`, and put that in `.gitignore`. This would be for credentials which are the same in development and production, like mailgun.

My apologies if that's what people were suggesting and I didn't get it. :)

Thoughts?